From cd83391c72aab5ea3732e70f8ebabc32d6440a0b Mon Sep 17 00:00:00 2001
From: Harald Hoyer <harald@hoyer.xyz>
Date: Thu, 11 Jan 2018 12:30:01 +0100
Subject: [PATCH] Merge pull request #350 from yuwata/fsfreeze

dracut.sh: do not invoke fsfreeze on EFI System Partition
---
 dracut.sh | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/dracut.sh b/dracut.sh
index c857b624..f6a75996 100755
--- a/dracut.sh
+++ b/dracut.sh
@@ -1834,8 +1834,10 @@ command -v restorecon &>/dev/null && restorecon -- "$outfile"
 if ! sync "$outfile" 2> /dev/null; then
     dinfo "dracut: sync operation on newly created initramfs $outfile failed"
     exit 1
+fi
+
 # use fsfreeze only if we're not writing to /
-elif ! [ "$(stat -c %m -- "$outfile")" == "/" ]; then
+if [[ "$(stat -c %m -- "$outfile")" != "/" && "$(stat -f -c %T -- "$outfile")" != "msdos" ]]; then
     if ! $(fsfreeze -f $(dirname "$outfile") 2>/dev/null && fsfreeze -u $(dirname "$outfile") 2>/dev/null); then
         dinfo "dracut: warning: could not fsfreeze $(dirname "$outfile")"
     fi

